### DISCUSSION TITLE
How to Add Interactive Map to Html  Getting Started with Reprocucible Science Project Page.

### DISCUSSION THREAD

**Discussion by oye-bobs** [cu-esiil-edu/2024-data-short-course#66] [Q&A] [answered]
@nquarder I just finished uploading my NOAA project html file to my portfolio page, but I realised that adding an interactive map to the project page would be very nice. I have generated an interactive map,ad downloaded it as an html file. How do I display it on my project page?

**Comment by nquarder** [comment]
Hi @oye-bobs please read these instructions as they will help you add that interactive map/html to your page - 

https://cu-esiil-edu.github.io/esiil-learning-portal/shortcourse/pages/03-git-github/03-github-portfolio/05-map.html

See the section **Download your map from Codespaces**

**Reply by oye-bobs** [reply]
I've done that. The problem is embedding it in the html file downloaded from the Getting started with reproducible science project codespace.

**Reply by nquarder** [reply]
Hey again @oye-bobs - I'm looking at your portfolio page (https://oye-bobs.github.io/) - and it looks great! And I see the map of Murtala Muhammed Airport in the section where you added the html file to your NOAA project notebook. 

https://oye-bobs.github.io/html/murtala_time_series.html

I think it looks awesome!

Are you trying to add something else? Maybe I'm missing something but from what I can tell you have this workflow figured out. 

One thing I did notice is the titles of your plots at the bottom of the notebook seem incorrect? 

For example:

![image](https://github.com/cu-esiil-edu/2024-data-short-course/assets/16786791/08a0a2b8-6b15-49fc-b87c-08cba1d15fc9)

The data look correct but the titles I think are incorrect for this location?

**Reply by oye-bobs** [reply]
Yeah, I want to add the map to the "murtala_time_series.html" file so that when the project is opened up, users can see the map.


And about the tiles, I didn't really know what to do.

how can I fix them?

**Comment by nquarder** [comment]
Hey @oye-bobs - what if you tried:

-from your murtala .ipynb ....

-delete the markdown cell w/ the html to embed your interactive map that I had you add today during office hours 

-add a code cell that calls for the murtala airport from osm

-add a code cell that creates and displays the interactive map 

-rerun all cells in the notebook 

-see if that gets the map to display in your new murtala .html 

Try that! @eculler was suggesting this but I was distracted and don't think it was sinking in for me. 😂

**Reply by oye-bobs** [reply]
Hey @nquarder , So I tried to add a code cell that calls for the murtala airport from osm, but I had to import the necessary libraries required for me to call for the murtala airport.

while trying to import those libraries, 
![5](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/e41eedcb-5b4c-45d4-a9b6-5f937033c3a3)


I got this error
![6](https://github.com/cu-esiil-edu/2024-data-short-course/assets/33554816/d1c4b172-07e5-46ed-8c80-68fd1b0126d3)

**Reply by nquarder** [reply, marked as answer]
okay @oye-bobs you will need to use the environment that @eculler created for the **first-map-template** repo if you want to add that interactive map to your Murtala analysis. 

you can do this by changing one line in the .devcontainer.json file you are using for your Murtala analysis.

- open up your Murtala analysis in a codepsace

- from the codespace, open the .devcontainer.json file

- change line 3 in your .devcontainer.json file from: 

`"registry.hub.docker.com/earthlab/earth-analytics-dev-env:latest"`

to

`"registry.hub.docker.com/earthlab/first-map:latest"`

You'll then be prompted to rebuild your codespace. Do this and then re-run the notebook from the top. 

I just tried this for my Cedar Rapids analysis and was able to get it to work: https://nquarder.github.io/notebooks/01-time-series-cedar-rapids.html

**Reply by oye-bobs** [reply]
Hi @nquarder , This worked, and I was able to add the interactive map. Thanks a lot.
